frontend/app/work_packages/directives/work-package-comment-directive.js

```diff
@@ -0,0 +1,153 @@
+    var ctrl = this;
+    ctrl.state = EditableFieldsState;
+    ctrl.field = 'activity-comment';
+
+    ctrl.editTitle = I18n.t('js.inplace.button_edit', { attribute: I18n.t('js.label_comment') });
+    ctrl.placeholder = I18n.t('js.label_add_comment_title');
+    ctrl.title = I18n.t('js.label_add_comment_title');
+
+    ctrl.state.isBusy = false;
+    ctrl.isEditing = ctrl.state.forcedEditState;
+    ctrl.canAddComment = !!ctrl.workPackage.links.addComment;
+
+    ctrl.showAbove = ConfigurationService.commentsSortedInDescendingOrder(); 
+
+    ctrl.isEmpty = function() {
+      return ctrl.writeValue === undefined || !ctrl.writeValue.raw;
+    };
+
+    ctrl.isEditable = function() {
+      return true;
+    };
+
+    ctrl.submit = function(notify) {
+      if (ctrl.isEmpty()) {
+        return;
+      }
+
+      ctrl.state.isBusy = true;
+      ActivityService.createComment(
+        ctrl.workPackage,
+        ctrl.writeValue,
+        notify
+      ).then(function(response) {
+        $scope.$emit('workPackageRefreshRequired', '');
+        ctrl.discardEditing();
+        return response;
+      }, function() {
+        NotificationsService.addError(I18n.t('js.comment_send_failed'));
+        ctrl.state.isBusy = false;
+      });
+    };
+
+    ctrl.startEditing = function(writeValue) {
+      ctrl.isEditing = true;
+      ctrl.writeValue = writeValue || { raw: '' };
+      ctrl.markActive();
+
+      $timeout(function() {
+        var inputElement = $element.find('.focus-input');
+        FocusHelper.focus(inputElement);
+        inputElement.triggerHandler('keyup');
+        ctrl.markActive();
+        inputElement.off('focus.inplace').on('focus.inplace', function() {
+          $scope.$apply(function() {
+            ctrl.markActive();
+          });
+        });
+      });
+    };
+
+    ctrl.discardEditing = function() {
+      delete ctrl.writeValue;
+      ctrl.isEditing = false;
+      ctrl.state.isBusy = false;
+    };
+
+    ctrl.isActive = function() {
+      if (EditableFieldsState.forcedEditState) {
+        return false;
+      }
+      return ctrl.field === EditableFieldsState.activeField;
+    };
+
+    ctrl.markActive = function() {
+      EditableFieldsState.activeField = ctrl.field;
+    };
+
+    if (!EditableFieldsState.forcedEditState) {
+      $element.bind('keydown keypress', function(e) {
+        if (e.keyCode === 27) {
+          $scope.$apply(function() {
+            ctrl.discardEditing();
+          });
+        }
+      });
+    }
+
+    $scope.$on('workPackage.comment.quoteThis', function(evt, quote) {
+      ctrl.startEditing({ raw: quote });
+    });
+  }
+
+  return {
+    restrict: 'E',
+    replace: true,
+    transclude: true,
+    controllerAs: 'fieldController',
+    bindToController: true,
+    templateUrl: '/templates/work_packages/comment.html',
+    scope: {
+      workPackage: '='
+    },
+    controller: commentFieldDirectiveController,
+    link: function(scope, element, attrs) {
```

Comment:
'attrs' is defined but never used.<br>'element' is defined but never used.<br>'scope' is defined but never used.